lib/elastic_apm/span.rb

```diff
@@ -98,13 +98,13 @@ def build_stacktrace
     end
 
     def long_enough_for_stacktrace?
-      min_duration = @stacktrace_builder.config.span_frames_min_duration
+      min_duration =
+        @stacktrace_builder.config.span_frames_min_duration * 1_000_000
 
-      case min_duration
-      when -1 then true
-      when 0 then false
-      else duration / 1000 >= min_duration
-      end
+      return true if min_duration < 0
+      return false if min_duration == 0
+
+      duration >= min_duration
```

Comment:
From the docs I understood that this should
```
return true if min_duration == -1
duration >= min_duration
```
Why the special case handling for `0`?

Reply:
That's how it's implemented in all agents. `0` means off, `-1` means always on.